### PR TITLE
[Feature] Support load json type from csv format through stream load

### LINKED ISSUE #9366: Support load json type from csv format through stream load
## Enhancement
origin_img_v2_4a865557-f943-460c-a2e6-2ced4949847g

### PR DESCRIPTION
**Pull request by smartlxh**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #9366

## Problem Summary(Required) ：

Support load json type from csv format through stream load by adding a json_converter

image

### CODE REVIEW

**Comment by imay** [comment]
@smartlxh should add some UT to cover new added code

File: be/src/formats/csv/converter.cpp [hunk 1]
```diff
@@ -62,6 +63,7 @@ static std::unique_ptr<Converter> get_converter(const TypeDescriptor& t) {
     case TYPE_TIME:
     case TYPE_OBJECT:
     case TYPE_JSON:
+        return std::make_unique<JsonConverter>();
```
Comment (mofeiatwork) [inline be/src/formats/csv/converter.cpp:66]: Above case should not return a `JsonConverter`.

**Comment by mofeiatwork** [comment]
run starrocks_admit_test

**Review by caneGuy — DISMISSED** [review]

File: be/src/formats/csv/json_converter.cpp [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+#include "common/logging.h"
+#include "gutil/casts.h"
+#include "types/date_value.hpp"
+
+namespace starrocks::vectorized::csv {
+
+Status JsonConverter::write_string(OutputStream* os, const Column& column, size_t row_num,
+                                   const Options& options) const {
+    auto data_column = down_cast<const JsonColumn*>(&column);
+    std::vector<JsonValue>& pool = const_cast<std::vector<JsonValue>&>(data_column->get_pool());
```
Comment (imay) [inline be/src/formats/csv/json_converter.cpp:15]: why const_cast?
```suggestion
    auto& pool = reinterpret_cast<std::vector<JsonValue>&>(data_column->get_pool());
```

File: be/src/formats/csv/json_converter.cpp [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    JsonValue b = pool.at(row_num);
+    return os->write(b.get_slice());
```
Comment (imay) [inline be/src/formats/csv/json_converter.cpp:17]:
```suggestion
    return os->write(pool[row_num].get_slice());
```
if you use JsonValue, you do a value copy. is this necessary?

**Review by imay — APPROVED** [review]
LGTM

**Review by caneGuy — APPROVED** [review]

**Comment by caneGuy** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by smartlxh** [comment]
run starrocks_admit_test

**Pull request merged**